src/Workspaces/CSharp/Portable/Execution/CSharpLanguageSpecificSerializationService.cs

```diff
@@ -0,0 +1,92 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Composition;
+using System.Linq;
+using System.Threading;
+using Microsoft.CodeAnalysis.Execution;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp.Execution
+{
+    [ExportLanguageService(typeof(ILanguageSpecificSerializationService), LanguageNames.CSharp), Shared]
+    internal class CSharpLanguageSpecificSerializationService : AbstractLanguageSpecificSerializationService
+    {
+        public override void WriteTo(CompilationOptions options, ObjectWriter writer, CancellationToken cancellationToken)
```

Comment:
can we add this functionality directly to the respective option types?  My concern here is that this decoupling means that every time we update those options we need to come here and fix up this code.